common/domainHash.js

```diff
@@ -0,0 +1,49 @@
+function byteToHex(byte) {
+  return ('0' + byte.toString(16)).slice(-2);
+}
+
+function generateId(len = 32) {
+  var arr = new Uint8Array(len / 2);
+  window.crypto.getRandomValues(arr);
+  return Array.from(arr, byteToHex).join("");
+}
+
+function gen_random32() {
+  var array = new Uint32Array(1);
+  window.crypto.getRandomValues(array);
+  return array[0];
+}
+
+function gen_random64() {
+  var array = new Uint32Array(2);
+  window.crypto.getRandomValues(array);
+  return BigInt("" + array[0] + array[1]);
+}
+
+if ((typeof browser) === "undefined") {
+  var browser = chrome;
+}
+
+function syncDomains(storageData) {
+  domains = storageData.visitedDomains;
+  sessionHash = storageData.sessionHash
+  if (!domains[location.origin]) {
+    domains[location.origin] = generateId();
+    chrome.storage.local.set({
+      "visitedDomains": domains
+    }, function() {
+
+    })
+
+
+  };
+  (function() {
+    var parent = document.documentElement;
+    var script = document.createElement('script');
+    script.async = false;
+    script.text = `var domainHash = "${domains[location.origin]}";var sessionHash ="${sessionHash}" `;
+    parent.insertBefore(script, parent.firstChild);
+    parent.removeChild(script);
```

Comment:
See injectScript() in inject.js

Anyway, what is the idea of this code? You polute global name space with variables `domainHash` and `sessionHash` which creates two problems:

* any page script can check if the variables are defined and based on this learn that the user is most likely using JSR,
* the page script can redefine the value of this variable.